stringvalidator/formats.go

```diff
@@ -0,0 +1,140 @@
+		description += "invalid configuration"
+	}
+
+	switch {
+	case len(validatorFormats.FormatsTypes) > 1:
+		description += "The value must respect the following rules : "
+	case len(validatorFormats.FormatsTypes) == 1:
+		description += "The value must respect the following rule : "
+	}
+
+	for i, formatsType := range validatorFormats.FormatsTypes {
+		if i == len(validatorFormats.FormatsTypes)-1 {
+			description += formatsTypesFunc[formatsType]().Description(ctx)
+		} else {
+			description += formatsTypesFunc[formatsType]().Description(ctx) + ", "
+		}
+	}
+
+	return description
+}
+
+// MarkdownDescription describes the validation in Markdown formatsting.
+func (validatorFormats formatsValidator) MarkdownDescription(ctx context.Context) string {
+	return validatorFormats.Description(ctx)
+}
+
+// Validate performs the validation.
+func (validatorFormats formatsValidator) ValidateString(
+	ctx context.Context,
+	request validator.StringRequest,
+	response *validator.StringResponse,
+) {
+	if request.ConfigValue.IsNull() || request.ConfigValue.IsUnknown() {
+		return
+	}
+
+	if len(validatorFormats.FormatsTypes) == 0 {
+		response.Diagnostics.AddAttributeError(
+			request.Path,
+			"No configuration provided",
+			fmt.Sprintf("invalid value: %s", request.ConfigValue.String()),
+		)
+		return
+	}
+
+	diags := diag.Diagnostics{}
+
+	// Check if the formats type is valid
+	// and call the validation function for each formats type
+	// and append the diagnostics to the response
+	// If the formats type is not valid, return an error
+	for _, formatsType := range validatorFormats.FormatsTypes {
+		d := new(validator.StringResponse)
+		if _, ok := formatsTypesFunc[formatsType]; !ok {
+			response.Diagnostics.AddError(
+				"Invalid formats type",
+				fmt.Sprintf("invalid formats type: %s", formatsType),
+			)
+			return
+		}
+		formatsTypesFunc[formatsType]().ValidateString(ctx, request, d)
+		diags.Append(d.Diagnostics...)
+	}
+
+	// Check if the value is valid for at least one of the formats types if comparatorOR is true
+	if validatorFormats.ComparatorOR && diags.ErrorsCount() == len(validatorFormats.FormatsTypes) {
+		response.Diagnostics.AddError(
+			fmt.Sprintf("Invalid configuration for attribute %s", request.Path),
+			"Set at least one valid formats type",
+		)
+	}
+
+	// If the value is valid for all of the formats types if comparatorOR is false (So AND logic)
+	if !validatorFormats.ComparatorOR {
+		response.Diagnostics.Append(diags...)
+	}
+}
+
+/*
+NewFormatsValidator creates a new FormatsValidator.
+It takes a list of formats types and a boolean comparatorOR.
+If comparatorOR is true, the value must be at least one of the formats types.
+If comparatorOR is false, the value must be all of the formats types.
+If no formats types are provided, the validator will return an error.
+If no comparatorOR is provided, it will default to false.
+The formats types are defined in the formatsTypes package.
```

Comment:
the package is not formatsTypes but formats. 

Rename formats package formatsTypes as for network